AI #1785: Add Invoice Detail and Billing Period datasets (#1900)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/invoice_detail/columns/paymentcurrency.md

# Payment Currency

> A Payment Currency represents the currency in which an invoice is paid, as determined by the invoice issuer. This currency may differ from the [Billing Currency](#datasets.costandusage.billingcurrency) if the customer and provider have agreed on a different settlement currency, making it essential for tracking actual cash outflow and foreign exchange impacts.

Reviewer: If my understanding is correct, this column (and `paymentCurrencyBilledCost` and `paymentCurrencyInvoiceDetailId`) is intended to represent the currency in which the invoice issuer is expecting to be paid in. This currency is distinct from the  currency underlying a payor's bank account or funding mechanism. I didn't read this to capture that 

Ex: Payor Inc. has USD-denominated account and incurs fees + fx markeup to wire EUR to pay on behalf of a subsidiary GmbH's EUR AWS invoice. 

**Accounting & finance personas understand the term "payment currency" in this case is USD, and there was an FX impact in order to fulfill the invoice in EUR.** If this dataset is foreseeably used to reconcile with accounting or bank data, the term "payment currency" may be difficult to understand alongside true accounting payment data and currencies. 

Suggested:
InvoiceCurrency
SettlementCurrency
uhpotatoCurrency

Author: @aqu-erp 
In your example, I think it still works.  The `PaymentCurrency` is EUR, and even though Payor Inc. has a USD-denominated account, the payment currency was indeed euros.

> Accounting & finance personas understand the term "payment currency" in this case is USD.

Tell me more about this.  The PDF invoice examples that we have for many data generators state the term "payment currency" clearly, and in many (if not all?) cases, this is accompanied by the term "preferred", which implies that Payor Inc. could tell the provider that they wish to pay in USD, thereby avoiding this collision.

I propose no change here, but I will leave this open to collect feedback from you and others.

Author: Discussed in Feb 17 TF-1.  Consensus not to change the column names.  However, there was some agreement that the terminology in the description should be finessed, especially as it pertains to the term "settlement".

```suggestion
A Payment Currency represents the currency in which an invoice is paid, as determined by the invoice issuer. This currency may differ from the [Billing Currency](#datasets.costandusage.billingcurrency) if the customer and provider have agreed on a different payment currency, allowing FinOps practitioners to track actual cash outflow and foreign exchange impacts.
```

Reviewer: ```suggestion
A Payment Currency represents the currency the invoice-issuer expects to receive. This currency may differ from the [Billing Currency](#datasets.costandusage.billingcurrency) if the customer and provider have agreed on a different settlement currency, making it essential for tracking actual cash outflow and foreign exchange impacts.
```

Adjusting for the fuzzy nature of how "is paid" could be interpreted from either payee or payor. Also not a hill I will die on if there is already consensus for the existing language.

Reviewer: Ah just missed the consensus update, got it.

Author: Combining your suggestion into a more holistic revision!  What do you think?

```suggestion
A Payment Currency represents the currency in which the invoice issuer requires settlement.  This is the currency of the financial obligation created by the invoice, which may differ from the [Billing Currency](#datasets.costandusage.billingcurrency) and/or the source currency of the payer’s funds or bank account.  Payment Currency allows FinOps practitioners to track settlement obligations and foreign exchange impacts.
```

## Requirements

> PaymentCurrency adheres to the following requirements:

Reviewer 2: ```suggestion
PaymentCurrency MUST adhere to the following requirements:
```
